x/examples/outline-connectivity/main.go

```diff
@@ -147,62 +164,48 @@ func main() {
 		resolverAddress := net.JoinHostPort(resolverHost, "53")
 		for _, proto := range strings.Split(*protoFlag, ",") {
 			proto = strings.TrimSpace(proto)
-			testTime := time.Now()
-			var testErr error
-			var testDuration time.Duration
+			var resolver dns.Resolver
 			switch proto {
 			case "tcp":
 				streamDialer, err := config.NewStreamDialer(*transportFlag)
 				if err != nil {
 					log.Fatalf("Failed to create StreamDialer: %v", err)
 				}
-				resolver := &transport.StreamDialerEndpoint{Dialer: streamDialer, Address: resolverAddress}
-				testDuration, testErr = connectivity.TestResolverStreamConnectivity(context.Background(), resolver, *domainFlag)
+				resolver = dns.NewTCPResolver(streamDialer, resolverAddress)
 			case "udp":
 				packetDialer, err := config.NewPacketDialer(*transportFlag)
 				if err != nil {
 					log.Fatalf("Failed to create PacketDialer: %v", err)
```

Comment:
On GUI apps, `log.Fatalf` leads to apps crashing on this error, however, it does not matter for `test_connectivity` example here. 

One solution I am using in pending mobile app PR [137](https://github.com/Jigsaw-Code/outline-sdk/pull/137) for GUI app (`outline-connectivity-app/shared_backed/main.go`) is to catch this error similar to other types of connectivity error (i.e protocol not supported by transport). 

Since no connection is made, I set the duration to `zero` seconds. This way the `go` backend can communicate the issue to the front-end through `ConnectivityTestResult` in response to `ConnectivityTestRequest` that contains the config parameters. This error gets collected to report collector similar to other connectivity errors. 

I think it makes sense to refactor the core logic of running a test such that the CLI, GUI app test operations are more consistent. I like the way the mobile app is implementing the logic with `ConnectivityTestRequest` and `ConnectivityTestResult`. Perhaps we can adopt that design in other places. 

Lastly the GUI app code also needs refactor to use `dns.NewUDPResolver` and `dns.NewTCPesolver`. I can do that through a separate PR.


```suggestion
			if err != nil {
				testDuration = time.Duration(0)
				testErr = err
			} else {
				resolver := &transport.PacketDialerEndpoint{Dialer: packetDialer, Address: resolverAddress}
				testDuration, testErr = connectivity.TestResolverPacketConnectivity(context.Background(), resolver, resolverAddress)
			}
```